src/cluster.c

```diff
@@ -5527,6 +5526,84 @@ void clusterReplyMultiBulkSlots(client * c) {
     setDeferredArrayLen(c, slot_replylen, num_masters);
 }
 
+sds genClusterInfoString(){
```

Comment:
i wonder if we can greatly reduce the diff by re-ordering things.
aside from the indentation issue, it also looks impossible to just move this function lower since it'll be in the middle of (not below) the original function. :disappointed:

Reply:
@oranagra I didnt really understand whats your comment here..

Reply:
i mean that since you moved a big pile of code into a separate function, the diff shows a lot of lines being deleted and a lot of lines being added.
this makes it harder to review, but also can cause unnecessary issues for cherry picks, and merges with some redis forks.
i generally try to avoid these when possible (sometimes even at an expense of unnecessary indentation).
sometimes i do that by adding a forward declaration, and then placing the function in the right place so that the diff is small.

in this case, i don't see how we can achieve that.
maybe the only way is to split `clusterCommand` to many small functions (placed according to the order of the `if-else` chain of the original one), and then implement a new `clusterCommand` that only has the `if-else` chain, and calls the individual functions.
i think in this case it would do more harm than good, so i'm leaning towards accepting your change as is, but wanted to raise that concern, to see if someone has another suggestion.

Reply:
@oranagra Kk, thanks for the detailed explenation.